src/odemis/acq/test/flim_test.py

```diff
@@ -272,6 +279,101 @@ def test_setting_stream(self):
+                                                self.sft, self.tc_scanner)
+
+        # Ask full ROI -> should have full ROI
+        helper.roi.value = (0, 0, 1, 1)
+        helper.repetition.value = (256, 256)
+        numpy.testing.assert_almost_equal(helper.roi.value, (0, 0, 1, 1))
+        self.assertEqual(helper.repetition.value, (256, 256))
+
+        pxs256f = helper.pixelSize.value
+
+        # Ask half ROI -> should have half the repetition (and same pixel size)
+        helper.roi.value = (0.5, 0.5, 1, 1)
+        numpy.testing.assert_almost_equal(helper.roi.value, (0.5, 0.5, 1, 1))
+        self.assertEqual(helper.repetition.value, (128, 128))
+        self.assertAlmostEqual(helper.pixelSize.value, pxs256f)
+
+        # Move the ROI around -> no change in repetition/pixel size
+        helper.roi.value = (0.25, 0.25, 0.75, 0.75)
+        numpy.testing.assert_almost_equal(helper.roi.value, (0.25, 0.25, 0.75, 0.75))
+        self.assertEqual(helper.repetition.value, (128, 128))
+        self.assertAlmostEqual(helper.pixelSize.value, pxs256f)
+
+        # Increase the ROI by "a little" -> as the laser-mirror doesn't support
+        # just a little bit more repetition, there is in theory two options:
+        # * the repetition should stay the same and pixel size increase
+        # * the repetition increases, and pixel size decreases
+        # Currently the implementation does the second option
+        helper.roi.value = (0.20, 0.20, 0.80, 0.80)
+        numpy.testing.assert_almost_equal(helper.roi.value, (0.20, 0.20, 0.80, 0.80))
+        self.assertEqual(helper.repetition.value, (256, 256))
+        self.assertLess(helper.pixelSize.value, pxs256f)
+
+        # TODO: reduce the ROI
+
+        pxs256h = helper.pixelSize.value
+
+        # Same thing but only change in one dimension (typical, from the GUI)
+        # X is twice smaller => exactly twice less rep, that's easy
+        helper.roi.value = (0.20, 0.20, 0.50, 0.80)
+        numpy.testing.assert_almost_equal(helper.roi.value, (0.20, 0.20, 0.50, 0.80))
+        self.assertEqual(helper.repetition.value, (128, 256))
+        self.assertAlmostEqual(helper.pixelSize.value, pxs256h)
+
+        # Slightly decrease X
+        # -> rep stays the same, pxs stays the same.
+        # ROI adjusted slightly on X (ideally it wouldn't), but size is the same
+        helper.roi.value = (0.20, 0.20, 0.45, 0.80)
+        self.assertEqual(helper.repetition.value, (128, 256))
+        self.assertAlmostEqual(helper.pixelSize.value, pxs256h)
+        roi = helper.roi.value
+        roi_size = (roi[2] - roi[0], roi[3] - roi[1])
+        numpy.testing.assert_almost_equal(roi_size, (0.3, 0.6))
+
+        # Move (back) => everything as requested
+        helper.roi.value = (0.20, 0.20, 0.50, 0.80)
+        numpy.testing.assert_almost_equal(helper.roi.value, (0.20, 0.20, 0.50, 0.80))
+
+        # Slightly increase X
```

Comment:
A bit confusing compared to the testcase before where you slightly decrease on X dim (only 0.05). There you say it should not affect the ROI size. In the testcase here you actually change the ROI size by 0.1 which should cause a change. Maybe rewrite that comment here, that you actually change X dim with the intention to change the rep value?